app/src/androidTest/java/com/github/se/signify/ui/screens/challenge/ChronoChallengeGameScreenTest.kt

```diff
@@ -149,4 +150,114 @@ class ChronoChallengeGameScreenTest {
+            round = 1,
+            roundWords = listOf("apple", "banana", "cherry"),
+            player1RoundCompleted = listOf(true, true, false),
+            player2RoundCompleted = listOf(false, false, false),
+            gameStatus = "in_progress")
+
+    // Act
+    val result = findNextIndex(challenge, "user1")
+
+    // Assert
+    assertEquals(2, result) // The first incomplete round for player1 is at index 2
+  }
+
+  @Test
+  fun findNextIndexReturnsCorrectIndexForPlayer2() {
+    // Arrange
+    val challenge =
+        Challenge(
+            challengeId = "testChallenge",
+            player1 = "user1",
+            player2 = "user2",
+            mode = "Chrono",
+            round = 1,
+            roundWords = listOf("apple", "banana", "cherry"),
+            player1RoundCompleted = listOf(true, true, false),
+            player2RoundCompleted = listOf(false, true, true),
+            gameStatus = "in_progress")
+
+    // Act
+    val result = findNextIndex(challenge, "user2")
+
+    // Assert
+    assertEquals(0, result) // The first incomplete round for player2 is at index 0
+  }
+
+  @Test
+  fun findNextIndexReturnsMinus1whenAllRoundsAreCompletedForPlayer1() {
+    // Arrange
+    val challenge =
+        Challenge(
+            challengeId = "testChallenge",
+            player1 = "user1",
+            player2 = "user2",
+            mode = "Chrono",
+            round = 1,
+            roundWords = listOf("apple", "banana", "cherry"),
+            player1RoundCompleted = listOf(true, true, true),
+            player2RoundCompleted = listOf(false, false, false),
+            gameStatus = "in_progress")
+
+    // Act
+    val result = findNextIndex(challenge, "user1")
+
+    // Assert
+    assertEquals(-1, result) // All rounds for player1 are completed
+  }
+
+  @Test
+  fun findNextIndexReturnsMinus1ForInvalidUser() {
+    // Arrange
+    val challenge =
+        Challenge(
+            challengeId = "testChallenge",
+            player1 = "user1",
+            player2 = "user2",
+            mode = "Chrono",
+            round = 1,
+            roundWords = listOf("apple", "banana", "cherry"),
+            player1RoundCompleted = listOf(false, false, false),
+            player2RoundCompleted = listOf(false, false, false),
+            gameStatus = "in_progress")
+
+    // Act
+    val result = findNextIndex(challenge, "invalidUser")
+
+    // Assert
+    assertEquals(0, result) // Invalid user should return -1
+  }
+
+  @Test
+  fun findNextIndexReturnsMinus1ForEmptyCompletedList() {
+    // Arrange
+    val challenge =
+        Challenge(
+            challengeId = "testChallenge",
+            player1 = "user1",
+            player2 = "user2",
+            mode = "Chrono",
+            round = 1,
+            roundWords = listOf("apple", "banana", "cherry"),
+            player1RoundCompleted = emptyList(),
+            player2RoundCompleted = emptyList(),
+            gameStatus = "in_progress")
+
```

Comment:
You should put this value in your setUp function since it's repeated multiple times.

Reply:
DONE